src/main/java/org/openstreetmap/atlas/utilities/command/subcommands/AtlasSearchCommand.java

```diff
@@ -0,0 +1,456 @@
+    private static final String INEDGE_OPTION_HINT = "ids";
+
+    private static final String OUTEDGE_OPTION_LONG = "outEdge";
+    private static final String OUTEDGE_OPTION_DESCRIPTION = "A comma separated list of out edge identifiers for which to search.";
+    private static final String OUTEDGE_OPTION_HINT = "ids";
+
+    private static final String PARENTRELATIONS_OPTION_LONG = "parentRelations";
+    private static final String PARENTRELATIONS_OPTION_DESCRIPTION = "A comma separated list of parent relation identifiers for which to search.";
+    private static final String PARENTRELATIONS_OPTION_HINT = "ids";
+
+    private static final String ID_OPTION_LONG = "id";
+    private static final String ID_OPTION_DESCRIPTION = "A comma separated list of Atlas ids for which to search.";
+    private static final String ID_OPTION_HINT = "ids";
+
+    private static final String OSMID_OPTION_LONG = "osmid";
+    private static final String OSMID_OPTION_DESCRIPTION = "A comma separated list of OSM ids for which to search.";
+    private static final String OSMID_OPTION_HINT = "osmids";
+
+    private static final String OUTPUT_ATLAS = "collected-multi.atlas";
+    private static final String COLLECT_OPTION_LONG = "collect-matching";
+    private static final String COLLECT_OPTION_DESCRIPTION = "Collect all matching atlas files and save to a file using the MultiAtlas.";
+
+    private static final Integer ALL_TYPES_CONTEXT = 3;
+    private static final Integer EDGE_ONLY_CONTEXT = 4;
+    private static final Integer NODE_ONLY_CONTEXT = 5;
+
+    private Set<String> wkts;
+    private TaggableFilter taggableFilter;
+    private Set<Long> startNodeIds;
+    private Set<Long> endNodeIds;
+    private Set<Long> inEdgeIds;
+    private Set<Long> outEdgeIds;
+    private Set<Long> parentRelations;
+
+    private Set<Long> ids;
+    private Set<Long> osmIds;
+    private Set<ItemType> typesToCheck;
+
+    private Set<Atlas> matchingAtlases;
+
+    private final OptionAndArgumentDelegate optionAndArgumentDelegate;
+    private final CommandOutputDelegate outputDelegate;
+
+    public static void main(final String[] args)
+    {
+        new AtlasSearchCommand().runSubcommandAndExit(args);
+    }
+
+    public AtlasSearchCommand()
+    {
+        this.optionAndArgumentDelegate = this.getOptionAndArgumentDelegate();
+        this.outputDelegate = this.getCommandOutputDelegate();
+        this.startNodeIds = new HashSet<>();
+        this.endNodeIds = new HashSet<>();
+        this.inEdgeIds = new HashSet<>();
+        this.outEdgeIds = new HashSet<>();
+        this.parentRelations = new HashSet<>();
+    }
+
+    @Override
+    public int finish()
+    {
+        if (this.optionAndArgumentDelegate.hasOption(COLLECT_OPTION_LONG)
+                && !this.matchingAtlases.isEmpty())
+        {
+            final Atlas outputAtlas = new MultiAtlas(this.matchingAtlases);
```

Comment:
Here skip the MultiAtlas step if the matching atlas set is only one element

Reply:
Updated! I figured leaving it this way made the code a bit simpler at some potential performance costs.